### PR TITLE
Refactored charts to use individual modules

### PR DESCRIPTION
**Pull request by dlabrecq**
Similar to the refactored list modules, this change allows each chart to be imported individually.

The user can now import individual modules like so:

```
import { DonutChartModule } from 'patternfly-ng/chart';
import { SparklineChartModule } from 'patternfly-ng/chart';
import { UtilizationDonutChartModule } from 'patternfly-ng/chart';
```

Or

```
import { 
  DonutChartModule, 
  SparklineChartModule, 
  UtilizationDonutChartModule 
} from 'patternfly-ng';
```

Example:
https://rawgit.com/dlabrecq/patternfly-ng/chart-modules-dist/dist-demo/#/utilization-donut

Fixes: https://github.com/patternfly/patternfly-ng/issues/345

Deprecations:
```
ChartModule  -- Use DonutChartModule or SparklineChartModule
DonutComponent  -- Use DonutChartComponent
DonutConfig  -- Use DonutChartConfig
SparklineComponent  -- Use SparklineChartComponent
SparklineConfig  -- Use SparklineChartConfig
SparklineData  -- Use SparklineChartData
```

### CODE REVIEW

File: src/app/card/basic-card/example/card-basic-example.component.html [hunk 1]
```diff
@@ -5,7 +5,7 @@
         <pfng-card [config]="config"
                    (onActionSelect)="handleActionSelect($event)"
                    (onFilterSelect)="handleFilterSelect($event)">
-          <pfng-chart-sparkline [config]="chartConfig" [chartData]="chartData" ></pfng-chart-sparkline>
+          <pfng-sparkline-chart [config]="chartConfig" [chartData]="chartData" ></pfng-sparkline-chart>
```
Comment (dtaylor113) [inline src/app/card/basic-card/example/card-basic-example.component.html:8]: Isn't this a Major breaking change?

File: src/app/card/basic-card/example/card-trend-example.component.html [hunk 2]
```diff
@@ -5,9 +5,9 @@
         <pfng-card [config]="config"
                    (onActionSelect)="handleActionSelect($event)"
                    (onFilterSelect)="handleFilterSelect($event)">
-          <pfng-chart-sparkline [config]="chartConfigVirtual" [chartData]="chartDataVirtual" ></pfng-chart-sparkline>
-          <pfng-chart-sparkline [config]="chartConfigPhysical" [chartData]="chartDataPhysical" ></pfng-chart-sparkline>
-          <pfng-chart-sparkline [config]="chartConfigMemory" [chartData]="chartDataMemory" ></pfng-chart-sparkline>
+          <pfng-sparkline-chart [config]="chartConfigVirtual" [chartData]="chartDataVirtual" ></pfng-sparkline-chart>
+          <pfng-sparkline-chart [config]="chartConfigPhysical" [chartData]="chartDataPhysical" ></pfng-sparkline-chart>
+          <pfng-sparkline-chart [config]="chartConfigMemory" [chartData]="chartDataMemory" ></pfng-sparkline-chart>
```
Comment (dtaylor113) [inline src/app/card/basic-card/example/card-trend-example.component.html:10]: Breaking change?

File: src/app/card/basic-card/example/card-basic-example.component.html (hunk 1 above)
Comment (dlabrecq) [reply src/app/card/basic-card/example/card-basic-example.component.html:8]: It's a deprecation in sparkline-chart.component.ts. Both tag names are available, but this conforms to the naming convention we've been using with our other components. Need to correct that before introducing several more charts.

File: src/app/card/basic-card/example/card-trend-example.component.html (hunk 2 above)
Comment (dlabrecq) [reply src/app/card/basic-card/example/card-trend-example.component.html:10]: It's a deprecation in sparkline-chart.component.ts -- both tag names are available.

**Comment by dlabrecq** [comment]
Checked-in with @dtaylor113 and this is good to go.

**Pull request merged**